controllers/core/builder_controller.go

```diff
@@ -160,19 +164,78 @@ func (r *BuilderReconciler) getBuilderResult(builder *openfunction.Builder, buil
 			return err
 		}
 
+		r.stopTimer(fmt.Sprintf("%s/%s", builder.Namespace, builder.Name))
 		log.V(1).Info("Update builder status", "state", res)
 	}
 
 	return nil
 }
 
+func (r *BuilderReconciler) startTimer(builder *openfunction.Builder) {
+	namespacedName := fmt.Sprintf("%s/%s", builder.Namespace, builder.Name)
+	log := r.Log.WithName("Timer").WithValues("Builder", namespacedName)
+
+	if builder.Spec.Timeout == nil ||
+		time.Since(builder.CreationTimestamp.Time) > builder.Spec.Timeout.Duration {
+		return
+	}
+
+	// Skipped when timer had started.
+	t, ok := r.timers[namespacedName]
+	if ok {
+		return
+	}
+
+	t = time.NewTimer(builder.Spec.Timeout.Duration - time.Since(builder.CreationTimestamp.Time))
+	r.timers[namespacedName] = t
```

Comment:
If the builder's configuration changes during the reconcile process, should the timer be retimed?

Reply:
If the builder's configuration changes, a new builder will be created, and a new timer will start. The old builder will be deleted.

Reply:
I see.